src/Compilers/CSharp/Test/Semantic/Semantics/RefFieldTests.cs

```diff
@@ -1124,7 +1124,10 @@ static void F([UnscopedRef] ref R1<int> r1)
             comp.VerifyEmitDiagnostics(
                 // (8,12): error CS9050: A ref field cannot refer to a ref struct.
                 //     public ref R1<T> F;
-                Diagnostic(ErrorCode.ERR_RefFieldCannotReferToRefStruct, "ref R1<T>").WithLocation(8, 12));
+                Diagnostic(ErrorCode.ERR_RefFieldCannotReferToRefStruct, "ref R1<T>").WithLocation(8, 12),
+                // (15,9): error CS9079: Cannot ref-assign 'r1' to 'F' because 'r1' can only escape the current method through a return statement.
+                //         r2.F = ref r1;
+                Diagnostic(ErrorCode.ERR_RefAssignReturnOnly, "r2.F = ref r1").WithArguments("F", "r1").WithLocation(15, 9));
```

Comment:
Had some ideas for tests we need to validate this change. I'm sure we have a lot of these already but was having trouble finding them all. Decided to just write out the core scenarios that I thought were impacted by this change. More than happy if these are already covered. 

https://gist.github.com/jaredpar/880869bcb54e38cb1e1475b1ff2543ef

Reply:
Looked around a bit and didn't feel like existing tests were quite similar enough. Added these tests in e582e59. I think some of the tests assumed that RTRS would have ReturnOnly scope by default. Also, `ReturnOnlyScope_03` has some cases where we assign a ref to a non-ref field, which I'm not sure was intentional.